### PR TITLE
Fixed format strings and type casts in hwclock to work with 64-bit time_t on 32-bit linux

### PR DESCRIPTION
**Pull request by arbego**
Fixes issue #1293

### CODE REVIEW

File: sys-utils/hwclock.c [hunk 1]
```diff
@@ -246,12 +246,21 @@ static int read_adjtime(const struct hwclock_control *ctl,
...
-	sscanf(line2, "%ld", &adjtime_p->last_calib_time);
+		sscanf(line2, "%lld", (int64_t*)&adjtime_p->last_calib_time);
+	} else {
+		sscanf(line1, "%lf %ld %lf",
+			&adjtime_p->drift_factor,
+			(long*)&adjtime_p->last_adj_time,
+			&adjtime_p->not_adjusted);
+
+		sscanf(line2, "%ld", (long*)&adjtime_p->last_calib_time);
+	}
```
Comment (karelzak) [inline sys-utils/hwclock.c:263]: Would be better to use locally only 64-bit numbers for scanf() and then cast to time_t ?
```C
int64_t last;

sscanf(line1, "%lf %"SCNs64"%lf", adjtime_p->drift_factor, &last, &adjtime_p->not_adjusted);
adjtime_p->last_adj_time = (time_t) last;
```
it seems less code. Note also SCNs64.

File: sys-utils/hwclock.c [hunk 2]
```diff
@@ -267,10 +276,10 @@ static int read_adjtime(const struct hwclock_control *ctl,
 
 	if (ctl->verbose) {
 		printf(_
-		       ("Last drift adjustment done at %ld seconds after 1969\n"),
-		       (long)adjtime_p->last_adj_time);
-		printf(_("Last calibration done at %ld seconds after 1969\n"),
-		       (long)adjtime_p->last_calib_time);
+		       ("Last drift adjustment done at %lld seconds after 1969\n"),
```
Comment (karelzak) [inline sys-utils/hwclock.c:279]: _("Last drift adjustment done at %"PRIs64" seconds after 1969\n").

There is also typo () -> _()

**Comment by karelzak** [comment]
Seems good. 

Please use macros (PRIs64 and SCNs64) from inttypes.h rather than %lld. It would be nice to also modify lib/timeutils.c where we also on some places assume that time_t is  "long int".

File: lib/timeutils.c [hunk 3]
```diff
@@ -504,7 +505,7 @@ int strtime_iso(const time_t *t, int flags, char *buf, size_t bufsz)
 	if (rc)
 		return format_iso_time(&tm, 0, flags, buf, bufsz);
 
-	warnx(_("time %ld is out of range."), (long)t);
+	warnx(_("time %"PRId64" is out of range."), (int64_t)*t);
```
Comment (arbego) [inline lib/timeutils.c:508]: Also fixed a potential bug where pointer address would be print instead of pointer content. Please confirm.

**Pull request merged**